### ISSUE TITLE
Make sure to grant all required permissions to owner of security definer functions

### ISSUE THREAD

**Issue by strk** [linz/linz-lds-bde-schema#64]
This schema defines a few SECURITY DEFINER functions.
This ticket is to make sure all operations performed by those SECURITY DEFINER can be performed by the enforced owner (definer) of those functions, via GRANT statements.

See also https://github.com/linz/linz-bde-schema/issues/62

**Comment by strk** [comment]
@palmerj regarding table_versiontables, as reported in https://github.com/linz/linz-bde-schema/issues/62#issuecomment-350054532, the error I'm getting now when bde_dba is not superuser is that `table_version.revision` cannot be edited (DELETE rows) by `bde_dba` user.

**Comment by strk** [comment]
The `table_version.revision` table is onwed by "linz", which was the user used to load the extension.
While BDE schema has code to give ownership of BDE schema items to `bde_dba` it does not do so with `table_version` tables.

Moreover, `table_version` usage is optional in `bde-schema` so it may be a good idea to grant those permissions here in lds schema

**Comment by strk** [comment]
Ok now I see that `sql/versioning/01-version_tables.sql` grants privileges to `bde_admin` rather than `bde_dba`, that's most likely part of the problem.

**Comment by strk** [comment]
Moreover, that script only grants permissions to specific table revision tables, not metadata tables as a whole

**Comment by strk** [comment]
The only two security definer functions are `bde_ext.LDS_MaintainAllFBDELayers` and `lds.LDS_MaintainSimplifiedLayers`. Both functions are owned by `bde_dba` rule so will run in his name.

We want to add tests for calling those two functions in a way that they actually do something.

**Comment by strk** [comment]
Writing tests for the two function requires the presence of `bde_control` schema. The schema is installed by `linz-bde-uploader`. Sounds like a circular dependency. For the purpose of this ticket (and #18) I'll try at mocking the missing functions

**Comment by strk** [comment]
The more functions I mock the more come out as required. I filed #94 for reducing this problem.
As for this ticket (review required permission) I know see that those two security definer functions will need access to the temporary schema created by the uploader. The uploader schema has no security definers so those temporary schemas will be owned by the user connecting during upload procedure. I found that the default configuration for linz-bde-uploader requests switching to the `bde_dba` rule as the first thing upon connect, so things should be fine (but it really depends on actual linz-bde-uploader configuration to retain this situation)
